static-assets/components/cstudio-common/common-api.js

```diff
@@ -9098,8 +9101,13 @@ CStudioAuthoring.FilesDiff = {
                 key;
 
             if(data && data["mime-type"]){
-                for( var i = 0; i < data["mime-type"].length; i++){
-                    confMimeType = data["mime-type"][i];
+                var mimeTypes = data["mime-type"];
+                if(!Array.isArray(mimeTypes)) {
+                    //support single values coming from SiteServiceImpl#createMap
+                    mimeTypes = [mimeTypes];
```

Comment:
This change is not required to support customization of component icons, but without it, you must add 2 or more mime-type configurations for the mime-type.xml file to be processed